cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -712,23 +732,66 @@ runBenchmark benchTracer
   -- TVar for termination.
   txSubmissionTerm :: MSTM.TVar IO Bool <- liftIO $ STM.newTVarIO False
 
-  liftIO $ do
-    txsLists <- STM.atomically $ STM.takeTMVar txsListsForTargetNodes
-    let targetNodesAddrsAndTxsLists = zip (NE.toList remoteAddresses) txsLists
-    allAsyncs <- forM targetNodesAddrsAndTxsLists $ \(remoteAddr, txsList) ->
-      -- Launch connection and submission threads for a peer
-      -- (corresponding to one target node).
-      launchTxPeer benchTracer
-                   benchmarkTracers
-                   txSubmissionTerm
-                   pInfoConfig
-                   localAddr
-                   remoteAddr
-                   updROEnv
-                   txsList
-    let allAsyncs' = intercalate [] [[c, s] | (c, s) <- allAsyncs]
-    -- Just wait for all threads to complete.
-    mapM_ (void . wait) allAsyncs'
+  txsLists <- liftIO $ STM.atomically $ STM.takeTMVar txsListsForTargetNodes
+  case explorerAPIEndpoint of
+    Nothing ->
+      -- There's no Explorer's API endpoint specified, submit transactions
+      -- to the target nodes via 'ouroboros-network'.
+      liftIO $ do
+        let targetNodesAddrsAndTxsLists = zip (NE.toList remoteAddresses) txsLists
+        allAsyncs <- forM targetNodesAddrsAndTxsLists $ \(remoteAddr, txsList) ->
+          -- Launch connection and submission threads for a peer
+          -- (corresponding to one target node).
+          launchTxPeer benchTracer
+                       benchmarkTracers
+                       txSubmissionTerm
+                       pInfoConfig
+                       localAddr
+                       remoteAddr
+                       updROEnv
+                       txsList
+        let allAsyncs' = intercalate [] [[c, s] | (c, s) <- allAsyncs]
+        -- Just wait for all threads to complete.
+        mapM_ (void . wait) allAsyncs'
+    Just (ExplorerAPIEnpoint endpoint) ->
+      -- Explorer's API endpoint is specified, submit transactions
+      -- to that endpoint using POST-request.
+      liftIO $ do
+        initialRequest <- parseRequest endpoint
+        txsList <- concat <$> mapM (STM.atomically . STM.takeTMVar) txsLists
+        submitTxsToExplorer benchTracer initialRequest txsList tpsRate
+
+submitTxsToExplorer
+  :: forall m.
+     ( m ~ IO
+     )
+  => Tracer IO (TraceBenchTxSubmit (Mempool.GenTxId ByronBlock))
+  -> Request
+  -> [GenTx ByronBlock]
+  -> TPSRate
+  -> m ()
```

Comment:
If `m ~ IO` then why not just replace `m` with `IO` ?

Reply:
Done.